httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheControlHeaderParser.java

```diff
@@ -0,0 +1,205 @@
+     * directives.
+     *
+     * @param header the header to parse
+     * @return a {@link CacheControlHeader} object containing the relevant caching directives
+     * @throws IllegalArgumentException if the header is {@code null}
+     * @throws MalformedCookieException if the value of the "max-age" or "s-maxage" field is malformed
+     */
+    public final CacheControlHeader parse(final Header header) throws MalformedCookieException {
+       Args.notNull(header, "Header");
+
+        final CacheControlHeader cacheControlHeader = new CacheControlHeader();
+        final CharArrayBuffer buffer;
+        final Tokenizer.Cursor cursor;
+        if (header instanceof FormattedHeader) {
+            buffer = ((FormattedHeader) header).getBuffer();
+            cursor = new Tokenizer.Cursor(((FormattedHeader) header).getValuePos(), buffer.length());
+        } else {
+            final String s = header.getValue();
+            if (s == null) {
+                return cacheControlHeader;
+            }
+            buffer = new CharArrayBuffer(s.length());
+            buffer.append(s);
+            cursor = new Tokenizer.Cursor(0, buffer.length());
+        }
+
+        while (!cursor.atEnd()) {
+            final String name = tokenParser.parseToken(buffer, cursor, TOKEN_DELIMS);
+            if (name.isEmpty()) {
+                continue;
+            }
+            if (cursor.atEnd()) {
+                return cacheControlHeader;
+            }
+            final int valueDelim = buffer.charAt(cursor.getPos());
+            cursor.updatePos(cursor.getPos() + 1);
+            if (valueDelim != EQUAL_CHAR) {
+                if (LOG.isWarnEnabled()) {
+                    LOG.warn("Header value is invalid: '{}'", header);
+                }
+                continue;
+            }
+            final String value = tokenParser.parseValue(buffer, cursor, VALUE_DELIMS);
+
+            if (!cursor.atEnd()) {
+                cursor.updatePos(cursor.getPos() + 1);
+            }
+
+            setHeader(cacheControlHeader, name, value);
+
+        }
+
+        return cacheControlHeader;
+    }
+
+    /**
+     * Sets the value of the specified Cache-Control header field in the given CacheControlHeader object.
+     * Returns true if the header field is not recognized, or if the value is malformed and throws a MalformedCookieException.
+     *
+     * @param cacheControlHeader the CacheControlHeader object to set the header field in
+     * @param name               the name of the Cache-Control header field to set
+     * @param value              the value of the Cache-Control header field to set
+     * @return true if the header field is not recognized
+     * @throws MalformedCookieException if the value is malformed and cannot be parsed as a long integer
+     */
+    private boolean setHeader(final CacheControlHeader cacheControlHeader, final String name, final String value) throws MalformedCookieException {
```

Comment:
@arturobernalg `MalformedCookieException` is clearly wrong here. This header has nothing to do with cookies. I also think the right thing to do here is to ignore values that cannot be interpreted as valid instead of throwing an exception. A log message at the DEBUG priority might be all right.

Another thing. Please  rename `CacheControlHeader` to `CacheControl` and make it nice and *immutable*. For that you may need to inline this method and create `CacheControl` instance at the last moment when exiting the `#parse` method.